File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> arr;
    void dfs(TreeNode cur, int level){
        if(cur==null)
            return;
        
        if(level>=arr.size())
            arr.add(0);
        dfs(cur.left, level+1);
        arr.set(level, cur.val);
        dfs(cur.right, level+1);
    }
    public List<Integer> rightSideView(TreeNode root) {
        arr = new ArrayList<>(); 
        dfs(root, 0);
        return arr;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* root, int lvl, vector<int>& sol){
        if(root==nullptr)
            return;
        
        if(lvl == sol.size())
            sol.push_back(root->val);
        
        dfs(root->right, lvl+1, sol);
        dfs(root->left, lvl+1, sol);
    }
    
    vector<int> rightSideView(TreeNode* root) {
        vector<int> sol;
        dfs(root, 0, sol);
        return sol;
    }
};